File: Web/Angular/src/app/interfaces/task.ts
import {Project} from './project';
import {List} from './list';
import {User} from './user';
import {Comment} from './comment';
import {Attachment} from './attachment';

export class Task {
  id: number;
  startDate: string;
  text: string;
  progress: number;
  duration: number;
  deadline: string;
  parent: number;
  name: string;
  description: string;
  status: number;
  statusText: string;
  priority: number;
  priorityText: string;
  finishedDate: string;
  createdDate: string;
  changedDate: string;
  createdBy: User;
  changedBy: User;
  project: Project;
  assignees: User[];
  list: List;
  comments: Comment[];
  effort: number;
  isArchived: boolean;
  attachments: Attachment[];
}

File: Entity/Models/Task.cs
using System;
using System.Collections.Generic;

namespace Entity.Models
{
    public class Task
    {
        public int ID { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public int CreatedBy { get; set; }
        public System.DateTime CreatedTime { get; set; }
        public Nullable<int> ChangedBy { get; set; }
        public Nullable<System.DateTime> ChangedTime { get; set; }
        public int Status { get; set; }
        public int Duration { get; set; }
        public int ListID { get; set; }
        public Nullable<int> Priority { get; set; }
        public Nullable<System.DateTime> StartDate { get; set; }
        public Nullable<System.DateTime> FinishedDate { get; set; }
        public int Effort { get; set; }
        public bool IsArchived { get; set; }

        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly")]
        public virtual ICollection<Attachment> Attachments { get; set; }
        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly")]
        public virtual ICollection<CheckList> CheckLists { get; set; }
        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly")]
        public virtual ICollection<Comment> Comments { get; set; }
        public virtual List List { get; set; }
        public virtual User User { get; set; }
        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly")]
        public virtual ICollection<TaskDependency> TaskDependencies { get; set; }
        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly")]
        public virtual ICollection<TaskDependency> TaskDependencies1 { get; set; }
        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly")]
        public virtual ICollection<UserTask> UserTasks { get; set; }

    }
}
